### ISSUE TITLE
the extension ESLint is configured as formatter but it cannot format "TypescriptReact"-files

### ISSUE THREAD

**Issue by stoplion** [microsoft/vscode-eslint#1413]
```
 "[typescript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint",
    "editor.formatOnSave": true,
    "editor.codeActionsOnSave": {
      "source.fixAll.eslint": true
    },
  },
  "[typescriptreact]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint",
    "editor.formatOnSave": true,
    "editor.codeActionsOnSave": {
      "source.fixAll.eslint": true
    },
  },
```

![image](https://user-images.githubusercontent.com/231343/152870048-e4d80bfa-c289-408c-b43c-65bbd6069d33.png)

**Comment by Byofa** [comment]
Same with vue and javascript files

**Comment by dbaeumer** [comment]
@stoplion can you please provider me with a Github repository I can clone that demos this. Otherwise very hard to tell what is going on since these since depend on many variables (configs, versions, ....)

**Comment by Byofa** [comment]
> Same with vue and javascript files

Ok it was my fault I had an error in my ``.eslintrc.js``, I didn't read carefully the eslint output.

**Comment by dbaeumer** [comment]
OK. Closing then.

**Comment by dgss** [comment]
Hi there, there is no log in the outputs but i'm having this issue. Any other potential reason for this?

**Comment by dbaeumer** [comment]
@dgss is ESLint activated when this happens (e.g. is there an output channel?)

**Comment by dgss** [comment]
@dbaeumer yes, but i've found my stupid reason. Since if I trigger 'Format document with' and select ESLint it is working, but when i trigger my shortcut to format the document with the default formatter I was getting this error. So just a remapping of the shortcut (removing the shortcut and resetting it back to the same shortcut) solved it. Maybe it helps someone else.

**Comment by dbaeumer** [comment]
@dgss thanks for pointing this out. Indeed helpful.

**Comment by Joaquim-Frances** [comment]
After the last vcCode update of March I see the this error again. Tryed to remap the key but not helped.

**Comment by adrielTosi** [comment]
Also see this error happening. It's funny because it works at one moment and then it stops at the next.

**Comment by johnc-ftl** [comment]
I was seeing this issue, and it was caused by outdated eslint extension config. I had:

```json
  "eslint.options": {
    "extensions": [
      ".ts",
      ".html",
      ".mjs"
    ]
  },
```

And it turns out that `extensions` is no longer supported eslint config. If you use `extensions` or other obsoleted options in your eslint options it will break the extension's ability to format files.

(IMO it's a bug that obsolete options break formatting)

**Comment by johncrim** [comment]
I was seeing this issue, and it was caused by outdated eslint extension config. I had:

```json
  "eslint.options": {
    "extensions": [
      ".ts",
      ".html",
      ".mjs"
    ]
  },
```

And it turns out that `extensions` is no longer supported eslint config. If you use `extensions` or other obsoleted options in your eslint options it will break the extension's ability to format files.

The "fix" was to delete `extensions` from `eslint-options`.

(IMO it's a bug that obsolete options break formatting)

**Comment by sKopheK** [comment]
it's pity there's nothing in output that could lead you to solution

**Comment by theIvanChornyi** [comment]
works for me, just add in settings.json
 `  "eslint.validate": [
    "javascript",
    "typescript",
    "typescriptreact"
  ]`, also name has to be associated: `  "files.associations": {
    "*.jsx": "javascriptreact",
    "*.tsx": "typescriptreact",
    "*.rb": "ruby",
    "*.css": "css",
    "*.ts": "typescript",
    "*.ts ": "typescript"
  },
`
Found it in Change log
<img width="478" height="456" alt="Image" src="https://github.com/user-attachments/assets/4e5b0c53-b214-4a0e-8973-6262695fe064" />,

So final settings

<img width="478" height="456" alt="Image" src="https://github.com/user-attachments/assets/e55ea7e7-5e94-4964-b0fa-9080d9d309c1" />

**Comment by jesonxie** [comment]
I tried many ways to solve this problem: modifying VS Code settings.json, downgrading/reinstalling the ESLint plugin, running npm i eslint, replacing .eslintrc with eslint.config.mjs, and so on. 

Finally, I found that the file being formatted, src/lib/console.js, was in the .eslintignore file. 
```
node_modules
coverage
benchmark
fixtures
.vscode
.idea
docs
lib
```
remove it and everying is fine.
